### ISSUE TITLE
The client scripts don't work in custom instances

### ISSUE THREAD

**Issue by IonicaBizau** [jillix/flow-nodejs#264]
https://github.com/jillix/engine#custom-module

``` sh
ionicabizau@desktop:~/Documents/_engineApps$ ./eve/engine /home/ionicabizau/Documents/_engineApps//test 10002│ionicabizau@desktop:~/Documents/_engineApps$ l                                                               
 trace                                                                                                       │engine/         engine-stable/  ev-engine/    service-stable/  starter-johnny/    test/                      
{"name":"test","hostname":"desktop","pid":8064,"module":"engine","instance":"engine","level":50,"err":{"messa│engine-parser/  eve/            service-dev/  starter/         svg.cytoscape.js/  test.js                    
ge":"ENOENT, open '/home/ionicabizau/Documents/_engineApps/test/a.js'","name":"Error","stack":"Error: ENOENT,│ionicabizau@desktop:~/Documents/_engineApps$ cd test/                                                        
 open '/home/ionicabizau/Documents/_engineApps/test/a.js'\n    at Error (native)","code":"ENOENT"},"msg":"ENO│ionicabizau@desktop:~/Documents/_engineApps/test$ l                                                          
ENT, open '/home/ionicabizau/Documents/_engineApps/test/a.js'","time":"2015-07-30T08:33:03.030Z","v":0}      │composition/  public/  server/                                                                               
{"name":"test","hostname":"desktop","pid":8064,"module":"engine","instance":"engine","level":50,"err":{"messa│ionicabizau@desktop:~/Documents/_engineApps/test$ vim composition/                                           
ge":"ENOENT, open '/home/ionicabizau/Documents/_engineApps/test/a.js'","name":"Error","stack":"Error: ENOENT,│localhost.pub.json  root.json                                                                                
 open '/home/ionicabizau/Documents/_engineApps/test/a.js'\n    at Error (native)","code":"ENOENT"},"msg":"ENO│ionicabizau@desktop:~/Documents/_engineApps/test$ vim composition/root.json                                  
ENT, open '/home/ionicabizau/Documents/_engineApps/test/a.js'","time":"2015-07-30T08:33:03.036Z","v":0}      │ionicabizau@desktop:~/Documents/_engineApps/test$ cat composition/root.json                                  
/home/ionicabizau/Documents/_engineApps/eve/lib/request.js:68                                                │{                                                                                                            
                    url.query.w = path[1].indexOf('resource.') === 0 ? false :                               │    "roles": {                                                                                               
                                         ^                                                                   │        "*": true                                                                                            
TypeError: Cannot read property 'indexOf' of undefined                                                       │    },                                                                                                       
    at handler (/home/ionicabizau/Documents/_engineApps/eve/lib/request.js:68:42)                            │   "module": {                                                                                               
    at Server.module.exports (/home/ionicabizau/Documents/_engineApps/eve/lib/session.js:60:5)               │      "main": "server/a.js",                                                                                 
    at Server.emit (events.js:110:17)                                                                        │      "public": "public",                                                                                    
    at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:491:12)                                  │      "config": {},                                                                                          
    at HTTPParser.parserOnHeadersComplete (_http_common.js:111:23)                                           │      "flow": [],                                                                                            
    at Socket.socketOnData (_http_server.js:343:22)                                                          │      "client": {                                                                                            
    at Socket.emit (events.js:107:17)                                                                        │         "module": [                                                                                         
    at readableAddChunk (_stream_readable.js:163:16)                                                         │            "a.js"                                                                                           
    at Socket.Readable.push (_stream_readable.js:126:10)                                                     │         ]                                                                                                   
    at TCP.onread (net.js:538:20)                                                                            │      }                                                                                                      
ionicabizau@desktop:~/Documents/_engineApps$ ^C                                                              │   },                                                                                                        
ionicabizau@desktop:~/Documents/_engineApps$                                                                 │   "name": "root"                                                                                            
                                                                                                             │}                                                                                                            
                                                                                                             │ionicabizau@desktop:~/Documents/_engineApps/test$                                                            
                                                                                                             │                                                       
```

**Comment by ottiker** [comment]
@IonicaBizau Can you post the composition?

**Comment by IonicaBizau** [comment]
@adioo Yes, is the right side, in the code block (refresh the page because I :memo: the issue)

**Comment by ottiker** [comment]
Custom module paths must be absolute, since they are fetched from the repo's public folder.

``` js
{
    "client": {
       "module": [
          "/public/script.js",
          "//external/script.js"
       ]
    }
}
```

**Comment by IonicaBizau** [comment]
Yes! But the server should _not_ crash. It should show a nice error.

**Comment by ottiker** [comment]
Fixed in 86398de8782ff425f168cc9469a2a58799d30dda
